service/frontend/adminHandler.go

```diff
@@ -278,7 +279,16 @@ func (adh *AdminHandler) AddSearchAttributes(ctx context.Context, request *admin
 		SkipSchemaUpdate:      request.GetSkipSchemaUpdate(),
 	}
 
-	run, err := adh.sdkClient.ExecuteWorkflow(
+	sdkClient, err := adh.sdkClientFactory.NewSystemClient(adh.logger)
+	if err != nil {
+		adh.logger.Fatal(
+			"error getting system sdk client",
+			tag.Error(err),
+		)
+		return nil, err
+	}
```

Comment:
This snippet is repeated many times in this PR. If we want to enforce this and say that if there is no valid sdk client for system namespace, then crash the app, that is fine, but I think it is batter to capture that logic in one place. For example, we can change the NewSystemClient() to GetSystemClient() which cache the created system client as singleton, and crash the process if it cannot create one. That way, all consumer place just do: 
sdkClient := sdkClientFactory.GetSystemClient()
and no error handling is needed on the consumer call site.